pyrit/prompt_target/http_target.py

```
@@ -0,0 +1,202 @@
+    ) -> None:
+
+        super().__init__(memory=memory)
+        self.http_request = http_request
+        self.callback_function = callback_function
+        self.prompt_regex_string = prompt_regex_string
+        self.response_parse_key = response_parse_key
+
+    async def send_prompt_async(self, *, prompt_request: PromptRequestResponse) -> PromptRequestResponse:
+        """
+        Sends prompt to HTTP endpoint and returns the response
+        """
+
+        self._validate_request(prompt_request=prompt_request)
+        request = prompt_request.request_pieces[0]
+
+        header_dict, http_body, url, http_method = self.parse_raw_http_request()
+        re_pattern = re.compile(self.prompt_regex_string)
+
+        # Make the actual HTTP request:
+
+        # Checks if the body is a json object - this matters when we substitute in the prompt for the placeholder
+        try:
+            json.loads(http_body)
+            http_body_json = True
+        except (ValueError, json.JSONDecodeError):
+            http_body_json = False
+
+        # Add Prompt into URL (if the URL takes it)
+        if re.search(self.prompt_regex_string, url):
+            prompt_url_safe = urllib.parse.quote(
+                request.original_value
+            )  # by default doing URL encoding for prompts that go in URL
+            formatted_url = re_pattern.sub(prompt_url_safe, url)
+            self.url = formatted_url
+
+        # Add Prompt into request body (if the body takes it)
+        if re.search(self.prompt_regex_string, http_body):
+            if http_body_json:  # clean prompt of whitespace control characters to ensure still valid json
+                cleaned_prompt = re.sub(r"\s", " ", request.original_value)
+                formatted_http_body = re_pattern.sub(cleaned_prompt, http_body)
+            else:  # doesn't clean prompt, enters it all in
+                formatted_http_body = re_pattern.sub(request.original_value, http_body)
```

If you have a `JSONDecodeError` that sets `http_body_json` to `False`, wouldn't we expect to see a JSONDecodeError thrown when you send the request through `requests.request()`?

If we expect the body to be JSON, I would just throw the error earlier than sending the request. If there are cases where the body does not need to be JSON and still can be sent through successfully, then this may be fine. Just flagging this to make sure its tested :-)

---

The body doesn't need to be a JSON necessarily, this depends on the endpoint (ie for AOAI this is a json and BIC this is just a one line string which is not JSON formatted at all)

---

Okay, gotcha. In cases where the JSON is malformed I could see this throwing again when its sent through the request. 

When do we see a `ValueError` for `json.load(http_body)`?

---

if we have something like in BIC where we just have a string as the body ie q={PROMPT}&qs=ds

vs AOAI makes you format as a json object

---

I think this code block will be removed per Rich's feedback about not worrying about the format in the target, but rather in the converters. Resolving